algorithms/src/snark/varuna/varuna.rs

```diff
@@ -670,7 +670,13 @@ where
 
             let input_fields = public_inputs_i
                 .iter()
-                .map(|input| input.borrow().to_field_elements())
+                .map(|input| {
+                    let input = input.borrow().to_field_elements()?;
+                    if input.len() > input_domain.size() - 1 {
```

Comment:
is input domain size guaranteed to be non-zero?

Reply:
This let me down an interesting rabbit hole. Yes it is guaranteed to be non-zero because we're always passing in a public input. And the - 1 is no longer needed in this somewhat related PR: https://github.com/AleoHQ/snarkVM/pull/2268

Reply:
So is the -1 still needed here?

Reply:
Nope, merged in the `mainnet` branch and resolved the conflict.